turbo/jsonrpc/overlay_api.go

```diff
@@ -0,0 +1,625 @@
+		select {
+		case <-ctx.Done():
+			failed = ctx.Err()
+			break blockLoop
+		case jobs <- task:
+		}
+		idx++
+	}
+	close(jobs)
+	pend.Wait()
+
+	// If execution failed in between, abort
+	if failed != nil {
+		log.Error("[GetLogs]", "failed", failed)
+		return nil, failed
+	}
+
+	logs := []*types.Log{}
+	for idx := range results {
+		res := results[idx]
+		if res == nil {
+			continue
+		}
+		log.Debug("logs for", "blockNumber", res.BlockNumber, "len(Logs)", len(res.Logs), "Error", res.Error)
+		logs = append(logs, res.Logs...)
+	}
+	log.Debug("FINAL LOGS", "len(logs)", len(logs))
+	return logs, nil
+}
+
+func filterLogs(logs types.Logs, addresses []common.Address, topics [][]common.Hash) []*types.Log {
+	addrMap := make(map[common.Address]struct{}, len(addresses))
+	for _, v := range addresses {
+		addrMap[v] = struct{}{}
+	}
+	return logs.Filter(addrMap, topics)
+}
+
+func (api *OverlayAPIImpl) replayBlock(ctx context.Context, blockNum uint64, statedb *state.IntraBlockState, chainConfig *chain.Config, tx kv.Tx) ([]*types.Log, error) {
+	log.Debug("[replayBlock] begin", "block", blockNum)
+	var (
+		hash               common.Hash
+		replayTransactions types.Transactions
+		evm                *vm.EVM
+		blockCtx           evmtypes.BlockContext
+		txCtx              evmtypes.TxContext
+		overrideBlockHash  map[uint64]common.Hash
+		baseFee            uint256.Int
+	)
+
+	blockLogs := []*types.Log{}
+	overrideBlockHash = make(map[uint64]common.Hash)
+
+	blockNumber := rpc.BlockNumber(blockNum)
+	blockNum, hash, _, err := rpchelper.GetBlockNumber(rpc.BlockNumberOrHash{BlockNumber: &blockNumber}, tx, api.filters)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := api.blockWithSenders(tx, hash, blockNum)
+	if err != nil || block == nil {
+		return nil, err
+	}
+
+	replayTransactions = block.Transactions()
+	log.Debug("[replayBlock] replayTx", "length", len(replayTransactions))
+
+	parent := block.Header()
+
+	if parent == nil {
+		return nil, fmt.Errorf("block %d(%x) not found", blockNum, hash)
+	}
+
+	getHash := func(i uint64) common.Hash {
+		if hash, ok := overrideBlockHash[i]; ok {
+			return hash
+		}
+		hash, err := api._blockReader.CanonicalHash(ctx, tx, i)
+		if err != nil {
+			log.Debug("Can't get block hash by number", "number", i, "only-canonical", true)
+		}
+		return hash
+	}
+
+	if parent.BaseFee != nil {
+		baseFee.SetFromBig(parent.BaseFee)
+	}
+
+	blockCtx = evmtypes.BlockContext{
+		CanTransfer: core.CanTransfer,
+		Transfer:    core.Transfer,
+		GetHash:     getHash,
+		Coinbase:    parent.Coinbase,
+		BlockNumber: parent.Number.Uint64(),
+		Time:        parent.Time,
+		Difficulty:  new(big.Int).Set(parent.Difficulty),
+		GasLimit:    parent.GasLimit,
+		BaseFee:     &baseFee,
+	}
+
+	signer := types.MakeSigner(chainConfig, blockNum, blockCtx.Time)
+	rules := chainConfig.Rules(blockNum, blockCtx.Time)
+
+	timeoutMilliSeconds := int64(500000)
+
+	timeout := time.Millisecond * time.Duration(timeoutMilliSeconds)
```

Comment:
extra note, please if you do the flag make it a Duration Flag so we don't have this.